src/statistics.jl

```diff
@@ -0,0 +1,88 @@
+abstract type AbstractStatistic end
+
+struct Mean{D<:AbstractDomain} <: AbstractStatistic
+  domain::D
+  values::Dict{Symbol, Vector} 
+end
+
+function Mean(solution::SimulationSolution)
+  # output dictionary
+  values = Dict(var => mean(real) for (var, real) in solution.realizations)
+    
+  Mean{typeof(solution.domain)}(solution.domain, values)
+end
+
+mean(solution::SimulationSolution) = Mean(solution)
+
+struct Variance{D<:AbstractDomain} <: AbstractStatistic
+  domain::D
+  values::Dict{Symbol, Vector} 
+end
+
+function Variance(solution::SimulationSolution)
+  # output dictionary
+  values = Dict(var => var(real) for (var, real) in solution.realizations)
+    
+  Variance{typeof(solution.domain)}(solution.domain, values)
+end
+
+var(solution::SimulationSolution) = Variance(solution)
+
+struct Quantile{D<:AbstractDomain, P<:Real}
+  domain::D
+  p::P
+  values::Dict{Symbol, Vector}
+end
+
+function Quantile(solution::SimulationSolution, p::Real)
+  # function returning vector of slices
+  slices(x) = [getindex.(x, i) for i in 1:npoints(solution.domain)]
+  
+  # output dictionary
+  values = Dict(var => quantile.(slices(real), p) for (var, real) in solution.realizations)
+    
+  Quantile{typeof(solution.domain), typeof(p)}(solution.domain, p, values)
+end
+
+quantile(solution::SimulationSolution, p::Real) = Quantile(solution, p)
+
+quantile(solution::SimulationSolution, ps::AbstractArray) = [quantile(solution, p) for p in ps]
+
+
+# ------------
+# IO methods
```

Comment:
Notice that all IO methods are the same except that they are dispatching on specific subtypes. You can replace all of them by a single method that takes a `AbstractStatistic` instead and Julia will take care of the rest.

Please print `Quantile` like the other statistics, no need to print the value of `p`. I am also not sure we need to store `p` in the object, it seems not that useful.